pkg/container/cri/json.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cri
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+/*
+Custom JSON / yaml (by way of json) serialization for these types
+*/
+
+func (m *Mount) MarshalJSON() ([]byte, error) {
+	type Alias Mount
+	name, ok := MountPropagation_name[int32(m.Propagation)]
+	if !ok {
+		return nil, fmt.Errorf("unknown propogation value: %v", m.Propagation)
+	}
+	return json.Marshal(&struct {
+		Propagation string `json:"propagation"`
+		*Alias
+	}{
+		Propagation: name,
+		Alias:       (*Alias)(m),
+	})
+}
+
+func (m *Mount) UnmarshalJSON(data []byte) error {
+	type Alias Mount
+	aux := &struct {
+		Propagation string `json:"propagation"`
+		*Alias
+	}{
+		Alias: (*Alias)(m),
+	}
+	if err := json.Unmarshal(data, &aux); err != nil {
```

Comment:
ideally everything "unmarshall" should be using [DisallowUnknownFields](https://gist.github.com/bndw/a95e49c8bb5581aa4685d289ec9a3aeb)
but i leave this to you.

Reply:
Yeah, trying to figure out if we should just do that inline or if there's a way to plumb through the option 🙄

Reply:
I'm going to tackle this in a follow up, but before the next release, I'm not sure how this will interact with the rest of the decoding logic.

i'm still hopeful for getting support for that upstream in API machinery.